  - 'extend-per-file-ignores' -> 'lint.extend-per-file-ignores'
warning: `one-blank-line-before-class` (D203) and `no-blank-line-before-class` (D211) are incompatible. Ignoring `one-blank-line-before-class`.
minitorch/__init__.py:1:1: D104 Missing docstring in public package
Found 1 error.

ruff-format..............................................................Failed
- hook id: ruff-format
- files were modified by this hook

warning: The top-level linter settings are deprecated in favour of their counterparts in the `lint` section. Please update the following options in `pyproject.toml`:
  - 'fixable' -> 'lint.fixable'
  - 'ignore' -> 'lint.ignore'
  - 'select' -> 'lint.select'
  - 'unfixable' -> 'lint.unfixable'
  - 'extend-per-file-ignores' -> 'lint.extend-per-file-ignores'
warning: `one-blank-line-before-class` (D203) and `no-blank-line-before-class` (D211) are incompatible. Ignoring `one-blank-line-before-class`.
1 file reformatted, 23 files left unchanged

pyright..................................................................Failed
- hook id: pyright
- exit code: 1

/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/tests/strategies.py
  /Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/tests/strategies.py:17:32 - error: "is_close" is not a known attribute of module "minitorch" (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations

(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pre-commit run --all 
check for added large files..............................................Passed
check for case conflicts.................................................Passed
check docstring is first.................................................Passed
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
check toml...............................................................Passed
debug statements (python)................................................Passed
mixed line ending........................................................Passed
fix requirements.txt.....................................................Passed
trim trailing whitespace.................................................Passed
ruff.....................................................................Passed
ruff-format..............................................................Passed
pyright..................................................................Failed
- hook id: pyright
- exit code: 1

/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/tests/strategies.py
  /Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/tests/strategies.py:17:32 - error: "is_close" is not a known attribute of module "minitorch" (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations

(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pre-commit run --all 
check for added large files..............................................Passed
check for case conflicts.................................................Passed
check docstring is first.................................................Passed
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
check toml...............................................................Passed
debug statements (python)................................................Passed
mixed line ending........................................................Passed
fix requirements.txt.....................................................Passed
trim trailing whitespace.................................................Passed
ruff.....................................................................Passed
ruff-format..............................................................Passed
pyright..................................................................Passed
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % python project/run_sentiment.py                           

Traceback (most recent call last):
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/project/run_sentiment.py", line 5, in <module>
    import minitorch
ModuleNotFoundError: No module named 'minitorch'
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % python project/run_sentiment.py

Traceback (most recent call last):
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/project/run_sentiment.py", line 5, in <module>
    import minitorch
ModuleNotFoundError: No module named 'minitorch'
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pre-commit run --all           
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pip install -e .
Obtaining file:///Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Installing backend dependencies ... done
  Preparing editable metadata (pyproject.toml) ... done
Building wheels for collected packages: minitorch
  Building editable for minitorch (pyproject.toml) ... done
  Created wheel for minitorch: filename=minitorch-0.5-py2.py3-none-any.whl size=930 sha256=d5f190ebf50687ddfe24f46f1e5933dc8e3ddc889ae24b7875500b14561682a5
  Stored in directory: /private/var/folders/pv/kccx0zjn2vbbz3svr1gczmt80000gn/T/pip-ephem-wheel-cache-gvww0jgi/wheels/d7/56/35/3bc5017be196a720e7c775c58e277452af2e578c6f64d615d5
Successfully built minitorch
Installing collected packages: minitorch
Successfully installed minitorch-0.5

[notice] A new release of pip is available: 23.1.2 -> 24.3.1
[notice] To update, run: pip install --upgrade pip
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % python project/run_sentiment.py

Traceback (most recent call last):
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/project/run_sentiment.py", line 6, in <module>
    from datasets import load_dataset
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/datasets/__init__.py", line 39, in <module>
    from .builder import ArrowBasedBuilder, BeamBasedBuilder, BuilderConfig, DatasetBuilder, GeneratorBasedBuilder
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/datasets/builder.py", line 85, in <module>
    @dataclass
     ^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 1230, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 1220, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 958, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 815, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'datasets.utils.version.Version'> for field version is not allowed: use default_factory
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pre-commit run --all           
check for added large files..............................................Passed
check for case conflicts.................................................Passed
check docstring is first.................................................Passed
check for merge conflicts................................................Passed
check for broken symlinks............................(no files to check)Skipped
check toml...............................................................Passed
debug statements (python)................................................Passed
mixed line ending........................................................Passed
fix requirements.txt.....................................................Passed
trim trailing whitespace.................................................Passed
ruff.....................................................................Passed
ruff-format..............................................................Passed
pyright..................................................................Passed
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % ls
README.md               minitorch               pyproject.toml          requirements.txt        sync_previous_module.py
files_to_sync.txt       project                 requirements.extra.txt  setup.py                tests
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % python project/run_sentiment.py

Traceback (most recent call last):
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/project/run_sentiment.py", line 6, in <module>
    from datasets import load_dataset
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/datasets/__init__.py", line 39, in <module>
    from .builder import ArrowBasedBuilder, BeamBasedBuilder, BuilderConfig, DatasetBuilder, GeneratorBasedBuilder
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/datasets/builder.py", line 85, in <module>
    @dataclass
     ^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 1230, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 1220, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 958, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 815, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'datasets.utils.version.Version'> for field version is not allowed: use default_factory
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pip install -r requirements.extra.txt

Requirement already satisfied: altair==4.2.2 in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 1)) (4.2.2)
Requirement already satisfied: datasets==2.4.0 in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 2)) (2.4.0)
Requirement already satisfied: embeddings==0.0.8 in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 3)) (0.0.8)
Requirement already satisfied: networkx==3.3 in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 4)) (3.3)
Requirement already satisfied: plotly==4.14.3 in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 5)) (4.14.3)
Requirement already satisfied: pydot==1.4.1 in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 6)) (1.4.1)
Requirement already satisfied: python-mnist in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 7)) (0.7)
Requirement already satisfied: streamlit==1.12.0 in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 8)) (1.12.0)
Requirement already satisfied: streamlit-ace in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 9)) (0.1.1)
Requirement already satisfied: torch in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 10)) (2.5.1)
Requirement already satisfied: watchdog==1.0.2 in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 11)) (1.0.2)
Requirement already satisfied: entrypoints in ./.venv/lib/python3.11/site-packages (from altair==4.2.2->-r requirements.extra.txt (line 1)) (0.4)
Requirement already satisfied: jinja2 in ./.venv/lib/python3.11/site-packages (from altair==4.2.2->-r requirements.extra.txt (line 1)) (3.1.4)
Requirement already satisfied: jsonschema>=3.0 in ./.venv/lib/python3.11/site-packages (from altair==4.2.2->-r requirements.extra.txt (line 1)) (4.23.0)
Requirement already satisfied: numpy in ./.venv/lib/python3.11/site-packages (from altair==4.2.2->-r requirements.extra.txt (line 1)) (2.0.0)
Requirement already satisfied: pandas>=0.18 in ./.venv/lib/python3.11/site-packages (from altair==4.2.2->-r requirements.extra.txt (line 1)) (2.2.3)
Requirement already satisfied: toolz in ./.venv/lib/python3.11/site-packages (from altair==4.2.2->-r requirements.extra.txt (line 1)) (1.0.0)
Requirement already satisfied: pyarrow>=6.0.0 in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (18.1.0)
Requirement already satisfied: dill<0.3.6 in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (0.3.5.1)
Requirement already satisfied: requests>=2.19.0 in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (2.32.3)
Requirement already satisfied: tqdm>=4.62.1 in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (4.67.1)
Requirement already satisfied: xxhash in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (3.5.0)
Requirement already satisfied: multiprocess in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (0.70.13)
Requirement already satisfied: fsspec[http]>=2021.11.1 in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (2024.10.0)
Requirement already satisfied: aiohttp in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (3.11.10)
Requirement already satisfied: huggingface-hub<1.0.0,>=0.1.0 in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (0.26.5)
Requirement already satisfied: packaging in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (24.2)
Requirement already satisfied: responses<0.19 in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (0.18.0)
Requirement already satisfied: retrying>=1.3.3 in ./.venv/lib/python3.11/site-packages (from plotly==4.14.3->-r requirements.extra.txt (line 5)) (1.3.4)
Requirement already satisfied: six in ./.venv/lib/python3.11/site-packages (from plotly==4.14.3->-r requirements.extra.txt (line 5)) (1.17.0)
Requirement already satisfied: pyparsing>=2.1.4 in ./.venv/lib/python3.11/site-packages (from pydot==1.4.1->-r requirements.extra.txt (line 6)) (3.2.0)
Requirement already satisfied: blinker>=1.0.0 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (1.9.0)
Requirement already satisfied: cachetools>=4.0 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (5.5.0)
Requirement already satisfied: click>=7.0 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (8.1.7)
Requirement already satisfied: importlib-metadata>=1.4 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (8.5.0)
Requirement already satisfied: pillow>=6.2.0 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (11.0.0)
Requirement already satisfied: protobuf<4,>=3.12 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (3.20.3)
Requirement already satisfied: pydeck>=0.1.dev5 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (0.9.1)
Requirement already satisfied: pympler>=0.9 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (1.1)
Requirement already satisfied: python-dateutil in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (2.9.0.post0)
Requirement already satisfied: rich>=10.11.0 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (13.9.4)
Requirement already satisfied: semver in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (3.0.2)
Requirement already satisfied: toml in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (0.10.2)
Requirement already satisfied: tornado>=5.0 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (6.4.2)
Requirement already satisfied: typing-extensions>=3.10.0.0 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (4.12.2)
Requirement already satisfied: tzlocal>=1.1 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (5.2)
Requirement already satisfied: validators>=0.2 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (0.34.0)
Requirement already satisfied: gitpython!=3.1.19 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (3.1.43)
Requirement already satisfied: filelock in ./.venv/lib/python3.11/site-packages (from torch->-r requirements.extra.txt (line 10)) (3.16.1)
Requirement already satisfied: sympy==1.13.1 in ./.venv/lib/python3.11/site-packages (from torch->-r requirements.extra.txt (line 10)) (1.13.1)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in ./.venv/lib/python3.11/site-packages (from sympy==1.13.1->torch->-r requirements.extra.txt (line 10)) (1.3.0)
Requirement already satisfied: aiohappyeyeballs>=2.3.0 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2)) (2.4.4)
Requirement already satisfied: aiosignal>=1.1.2 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2)) (1.3.1)
Requirement already satisfied: attrs>=17.3.0 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2)) (24.2.0)
Requirement already satisfied: frozenlist>=1.1.1 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2)) (1.5.0)
Requirement already satisfied: multidict<7.0,>=4.5 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2)) (6.1.0)
Requirement already satisfied: propcache>=0.2.0 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2)) (0.2.1)
Requirement already satisfied: yarl<2.0,>=1.17.0 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2)) (1.18.3)
Requirement already satisfied: gitdb<5,>=4.0.1 in ./.venv/lib/python3.11/site-packages (from gitpython!=3.1.19->streamlit==1.12.0->-r requirements.extra.txt (line 8)) (4.0.11)
Requirement already satisfied: pyyaml>=5.1 in ./.venv/lib/python3.11/site-packages (from huggingface-hub<1.0.0,>=0.1.0->datasets==2.4.0->-r requirements.extra.txt (line 2)) (6.0.2)
Requirement already satisfied: zipp>=3.20 in ./.venv/lib/python3.11/site-packages (from importlib-metadata>=1.4->streamlit==1.12.0->-r requirements.extra.txt (line 8)) (3.21.0)
Requirement already satisfied: jsonschema-specifications>=2023.03.6 in ./.venv/lib/python3.11/site-packages (from jsonschema>=3.0->altair==4.2.2->-r requirements.extra.txt (line 1)) (2024.10.1)
Requirement already satisfied: referencing>=0.28.4 in ./.venv/lib/python3.11/site-packages (from jsonschema>=3.0->altair==4.2.2->-r requirements.extra.txt (line 1)) (0.35.1)
Requirement already satisfied: rpds-py>=0.7.1 in ./.venv/lib/python3.11/site-packages (from jsonschema>=3.0->altair==4.2.2->-r requirements.extra.txt (line 1)) (0.22.3)
Requirement already satisfied: pytz>=2020.1 in ./.venv/lib/python3.11/site-packages (from pandas>=0.18->altair==4.2.2->-r requirements.extra.txt (line 1)) (2024.2)
Requirement already satisfied: tzdata>=2022.7 in ./.venv/lib/python3.11/site-packages (from pandas>=0.18->altair==4.2.2->-r requirements.extra.txt (line 1)) (2024.2)
Requirement already satisfied: MarkupSafe>=2.0 in ./.venv/lib/python3.11/site-packages (from jinja2->altair==4.2.2->-r requirements.extra.txt (line 1)) (3.0.2)
Requirement already satisfied: charset-normalizer<4,>=2 in ./.venv/lib/python3.11/site-packages (from requests>=2.19.0->datasets==2.4.0->-r requirements.extra.txt (line 2)) (3.4.0)
Requirement already satisfied: idna<4,>=2.5 in ./.venv/lib/python3.11/site-packages (from requests>=2.19.0->datasets==2.4.0->-r requirements.extra.txt (line 2)) (3.10)
Requirement already satisfied: urllib3<3,>=1.21.1 in ./.venv/lib/python3.11/site-packages (from requests>=2.19.0->datasets==2.4.0->-r requirements.extra.txt (line 2)) (2.2.3)
Requirement already satisfied: certifi>=2017.4.17 in ./.venv/lib/python3.11/site-packages (from requests>=2.19.0->datasets==2.4.0->-r requirements.extra.txt (line 2)) (2024.8.30)
Requirement already satisfied: markdown-it-py>=2.2.0 in ./.venv/lib/python3.11/site-packages (from rich>=10.11.0->streamlit==1.12.0->-r requirements.extra.txt (line 8)) (3.0.0)
Requirement already satisfied: pygments<3.0.0,>=2.13.0 in ./.venv/lib/python3.11/site-packages (from rich>=10.11.0->streamlit==1.12.0->-r requirements.extra.txt (line 8)) (2.18.0)
Requirement already satisfied: smmap<6,>=3.0.1 in ./.venv/lib/python3.11/site-packages (from gitdb<5,>=4.0.1->gitpython!=3.1.19->streamlit==1.12.0->-r requirements.extra.txt (line 8)) (5.0.1)
Requirement already satisfied: mdurl~=0.1 in ./.venv/lib/python3.11/site-packages (from markdown-it-py>=2.2.0->rich>=10.11.0->streamlit==1.12.0->-r requirements.extra.txt (line 8)) (0.1.2)

[notice] A new release of pip is available: 23.1.2 -> 24.3.1
[notice] To update, run: pip install --upgrade pip
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pip install -r requirements.txt      

Requirement already satisfied: colorama==0.4.3 in ./.venv/lib/python3.11/site-packages (from -r requirements.txt (line 1)) (0.4.3)
Requirement already satisfied: hypothesis==6.54 in ./.venv/lib/python3.11/site-packages (from -r requirements.txt (line 2)) (6.54.0)
Requirement already satisfied: numba==0.60 in ./.venv/lib/python3.11/site-packages (from -r requirements.txt (line 3)) (0.60.0)
Requirement already satisfied: numpy==2.0.0 in ./.venv/lib/python3.11/site-packages (from -r requirements.txt (line 4)) (2.0.0)
Requirement already satisfied: pre-commit==2.20.0 in ./.venv/lib/python3.11/site-packages (from -r requirements.txt (line 5)) (2.20.0)
Requirement already satisfied: pytest==8.3.2 in ./.venv/lib/python3.11/site-packages (from -r requirements.txt (line 6)) (8.3.2)
Requirement already satisfied: pytest-env in ./.venv/lib/python3.11/site-packages (from -r requirements.txt (line 7)) (1.1.4)
Requirement already satisfied: pytest-runner==5.2 in ./.venv/lib/python3.11/site-packages (from -r requirements.txt (line 8)) (5.2)
Requirement already satisfied: typing_extensions in ./.venv/lib/python3.11/site-packages (from -r requirements.txt (line 9)) (4.12.2)
Requirement already satisfied: attrs>=19.2.0 in ./.venv/lib/python3.11/site-packages (from hypothesis==6.54->-r requirements.txt (line 2)) (24.2.0)
Requirement already satisfied: sortedcontainers<3.0.0,>=2.1.0 in ./.venv/lib/python3.11/site-packages (from hypothesis==6.54->-r requirements.txt (line 2)) (2.4.0)
Requirement already satisfied: llvmlite<0.44,>=0.43.0dev0 in ./.venv/lib/python3.11/site-packages (from numba==0.60->-r requirements.txt (line 3)) (0.43.0)
Requirement already satisfied: cfgv>=2.0.0 in ./.venv/lib/python3.11/site-packages (from pre-commit==2.20.0->-r requirements.txt (line 5)) (3.4.0)
Requirement already satisfied: identify>=1.0.0 in ./.venv/lib/python3.11/site-packages (from pre-commit==2.20.0->-r requirements.txt (line 5)) (2.6.3)
Requirement already satisfied: nodeenv>=0.11.1 in ./.venv/lib/python3.11/site-packages (from pre-commit==2.20.0->-r requirements.txt (line 5)) (1.9.1)
Requirement already satisfied: pyyaml>=5.1 in ./.venv/lib/python3.11/site-packages (from pre-commit==2.20.0->-r requirements.txt (line 5)) (6.0.2)
Requirement already satisfied: toml in ./.venv/lib/python3.11/site-packages (from pre-commit==2.20.0->-r requirements.txt (line 5)) (0.10.2)
Requirement already satisfied: virtualenv>=20.0.8 in ./.venv/lib/python3.11/site-packages (from pre-commit==2.20.0->-r requirements.txt (line 5)) (20.28.0)
Requirement already satisfied: iniconfig in ./.venv/lib/python3.11/site-packages (from pytest==8.3.2->-r requirements.txt (line 6)) (2.0.0)
Requirement already satisfied: packaging in ./.venv/lib/python3.11/site-packages (from pytest==8.3.2->-r requirements.txt (line 6)) (24.2)
Requirement already satisfied: pluggy<2,>=1.5 in ./.venv/lib/python3.11/site-packages (from pytest==8.3.2->-r requirements.txt (line 6)) (1.5.0)
Requirement already satisfied: distlib<1,>=0.3.7 in ./.venv/lib/python3.11/site-packages (from virtualenv>=20.0.8->pre-commit==2.20.0->-r requirements.txt (line 5)) (0.3.9)
Requirement already satisfied: filelock<4,>=3.12.2 in ./.venv/lib/python3.11/site-packages (from virtualenv>=20.0.8->pre-commit==2.20.0->-r requirements.txt (line 5)) (3.16.1)
Requirement already satisfied: platformdirs<5,>=3.9.1 in ./.venv/lib/python3.11/site-packages (from virtualenv>=20.0.8->pre-commit==2.20.0->-r requirements.txt (line 5)) (4.3.6)

[notice] A new release of pip is available: 23.1.2 -> 24.3.1
[notice] To update, run: pip install --upgrade pip
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % python project/run_sentiment.py      

Traceback (most recent call last):
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/project/run_sentiment.py", line 6, in <module>
    from datasets import load_dataset
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/datasets/__init__.py", line 39, in <module>
    from .builder import ArrowBasedBuilder, BeamBasedBuilder, BuilderConfig, DatasetBuilder, GeneratorBasedBuilder
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/datasets/builder.py", line 85, in <module>
    @dataclass
     ^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 1230, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 1220, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 958, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 815, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'datasets.utils.version.Version'> for field version is not allowed: use default_factory
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pip install python-mnist
Requirement already satisfied: python-mnist in ./.venv/lib/python3.11/site-packages (0.7)

[notice] A new release of pip is available: 23.1.2 -> 24.3.1
[notice] To update, run: pip install --upgrade pip
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % mnist_get_data.sh
--2024-12-06 16:56:04--  http://yann.lecun.com/exdb/mnist/
Resolving yann.lecun.com (yann.lecun.com)... 216.165.22.6
Connecting to yann.lecun.com (yann.lecun.com)|216.165.22.6|:80... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://yann.lecun.com/exdb/mnist/ [following]
--2024-12-06 16:56:05--  https://yann.lecun.com/exdb/mnist/
Connecting to yann.lecun.com (yann.lecun.com)|216.165.22.6|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 29170 (28K) [text/html]
Saving to: ‘data/index.html.tmp’

index.html.tmp                           100%[================================================================================>]  28.49K  --.-KB/s    in 0.03s   

2024-12-06 16:56:05 (842 KB/s) - ‘data/index.html.tmp’ saved [29170/29170]

Loading robots.txt; please ignore errors.
--2024-12-06 16:56:05--  https://yann.lecun.com/robots.txt
Reusing existing connection to yann.lecun.com:443.
HTTP request sent, awaiting response... 404 Not Found
2024-12-06 16:56:05 ERROR 404: Not Found.

Loading robots.txt; please ignore errors.
--2024-12-06 16:56:05--  http://yann.lecun.com/robots.txt
Connecting to yann.lecun.com (yann.lecun.com)|216.165.22.6|:80... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://yann.lecun.com/robots.txt [following]
--2024-12-06 16:56:05--  https://yann.lecun.com/robots.txt
Connecting to yann.lecun.com (yann.lecun.com)|216.165.22.6|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2024-12-06 16:56:05 ERROR 404: Not Found.

Removing data/index.html.tmp since it should be rejected.

--2024-12-06 16:56:05--  http://yann.lecun.com/
Connecting to yann.lecun.com (yann.lecun.com)|216.165.22.6|:80... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://yann.lecun.com/ [following]
--2024-12-06 16:56:05--  https://yann.lecun.com/
Connecting to yann.lecun.com (yann.lecun.com)|216.165.22.6|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 69304 (68K) [text/html]
Saving to: ‘data/index.html.tmp’

index.html.tmp                           100%[================================================================================>]  67.68K  --.-KB/s    in 0.1s    

2024-12-06 16:56:05 (610 KB/s) - ‘data/index.html.tmp’ saved [69304/69304]

Removing data/index.html.tmp since it should be rejected.

--2024-12-06 16:56:05--  https://yann.lecun.com/exdb/mnist/train-images-idx3-ubyte.gz
Reusing existing connection to yann.lecun.com:443.
HTTP request sent, awaiting response... 403 Forbidden
2024-12-06 16:56:06 ERROR 403: Forbidden.

--2024-12-06 16:56:06--  https://yann.lecun.com/exdb/mnist/train-labels-idx1-ubyte.gz
Reusing existing connection to yann.lecun.com:443.
HTTP request sent, awaiting response... 403 Forbidden
2024-12-06 16:56:06 ERROR 403: Forbidden.

--2024-12-06 16:56:06--  https://yann.lecun.com/exdb/mnist/t10k-images-idx3-ubyte.gz
Reusing existing connection to yann.lecun.com:443.
HTTP request sent, awaiting response... 403 Forbidden
2024-12-06 16:56:06 ERROR 403: Forbidden.

--2024-12-06 16:56:06--  https://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz
Reusing existing connection to yann.lecun.com:443.
HTTP request sent, awaiting response... 403 Forbidden
2024-12-06 16:56:06 ERROR 403: Forbidden.

FINISHED --2024-12-06 16:56:06--
Total wall clock time: 1.2s
Downloaded: 2 files, 96K in 0.1s (664 KB/s)
~/Documents/Cornell/MLE/mod4-manzala/data ~/Documents/Cornell/MLE/mod4-manzala
gunzip: can't stat: * (*.gz): No such file or directory
~/Documents/Cornell/MLE/mod4-manzala
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % python project/run_sentiment.py

Traceback (most recent call last):
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/project/run_sentiment.py", line 6, in <module>
    from datasets import load_dataset
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/datasets/__init__.py", line 39, in <module>
    from .builder import ArrowBasedBuilder, BeamBasedBuilder, BuilderConfig, DatasetBuilder, GeneratorBasedBuilder
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/datasets/builder.py", line 85, in <module>
    @dataclass
     ^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 1230, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 1220, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 958, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 815, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'datasets.utils.version.Version'> for field version is not allowed: use default_factory
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pre-commit run --all                 
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % ls
README.md               files_to_sync.txt       project                 requirements.extra.txt  setup.py                tests
data                    minitorch               pyproject.toml          requirements.txt        sync_previous_module.py
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % deactivate
syedakazmi-shah@Syedas-MBP mod4-manzala % ls
README.md               files_to_sync.txt       project                 requirements.extra.txt  setup.py                tests
data                    minitorch               pyproject.toml          requirements.txt        sync_previous_module.py
syedakazmi-shah@Syedas-MBP mod4-manzala % rm -rf .venv 
syedakazmi-shah@Syedas-MBP mod4-manzala % cd /Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala
python3 -m venv .venv

syedakazmi-shah@Syedas-MBP mod4-manzala % source .venv/bin/activate

(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % mnist_get_data.sh                                           
data directory already present, exiting
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % source .venv/bin/activate

(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pip install python-mnist
Collecting python-mnist
  Using cached python_mnist-0.7-py2.py3-none-any.whl (9.6 kB)
Installing collected packages: python-mnist
Successfully installed python-mnist-0.7

[notice] A new release of pip is available: 23.1.2 -> 24.3.1
[notice] To update, run: pip install --upgrade pip
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pip3 install -r requirements.txt 
Collecting colorama==0.4.3 (from -r requirements.txt (line 1))
  Using cached colorama-0.4.3-py2.py3-none-any.whl (15 kB)
Collecting hypothesis==6.54 (from -r requirements.txt (line 2))
  Using cached hypothesis-6.54.0-py3-none-any.whl (389 kB)
Collecting numba==0.60 (from -r requirements.txt (line 3))
  Using cached numba-0.60.0-cp311-cp311-macosx_11_0_arm64.whl (2.6 MB)
Collecting numpy==2.0.0 (from -r requirements.txt (line 4))
  Using cached numpy-2.0.0-cp311-cp311-macosx_14_0_arm64.whl (5.2 MB)
Collecting pre-commit==2.20.0 (from -r requirements.txt (line 5))
  Using cached pre_commit-2.20.0-py2.py3-none-any.whl (199 kB)
Collecting pytest==8.3.2 (from -r requirements.txt (line 6))
  Using cached pytest-8.3.2-py3-none-any.whl (341 kB)
Collecting pytest-env (from -r requirements.txt (line 7))
  Using cached pytest_env-1.1.5-py3-none-any.whl (6.1 kB)
Collecting pytest-runner==5.2 (from -r requirements.txt (line 8))
  Using cached pytest_runner-5.2-py2.py3-none-any.whl (6.8 kB)
Collecting typing_extensions (from -r requirements.txt (line 9))
  Using cached typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Collecting attrs>=19.2.0 (from hypothesis==6.54->-r requirements.txt (line 2))
  Using cached attrs-24.2.0-py3-none-any.whl (63 kB)
Collecting sortedcontainers<3.0.0,>=2.1.0 (from hypothesis==6.54->-r requirements.txt (line 2))
  Using cached sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)
Collecting llvmlite<0.44,>=0.43.0dev0 (from numba==0.60->-r requirements.txt (line 3))
  Using cached llvmlite-0.43.0-cp311-cp311-macosx_11_0_arm64.whl (28.8 MB)
Collecting cfgv>=2.0.0 (from pre-commit==2.20.0->-r requirements.txt (line 5))
  Using cached cfgv-3.4.0-py2.py3-none-any.whl (7.2 kB)
Collecting identify>=1.0.0 (from pre-commit==2.20.0->-r requirements.txt (line 5))
  Using cached identify-2.6.3-py2.py3-none-any.whl (99 kB)
Collecting nodeenv>=0.11.1 (from pre-commit==2.20.0->-r requirements.txt (line 5))
  Using cached nodeenv-1.9.1-py2.py3-none-any.whl (22 kB)
Collecting pyyaml>=5.1 (from pre-commit==2.20.0->-r requirements.txt (line 5))
  Using cached PyYAML-6.0.2-cp311-cp311-macosx_11_0_arm64.whl (172 kB)
Collecting toml (from pre-commit==2.20.0->-r requirements.txt (line 5))
  Using cached toml-0.10.2-py2.py3-none-any.whl (16 kB)
Collecting virtualenv>=20.0.8 (from pre-commit==2.20.0->-r requirements.txt (line 5))
  Using cached virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
Collecting iniconfig (from pytest==8.3.2->-r requirements.txt (line 6))
  Using cached iniconfig-2.0.0-py3-none-any.whl (5.9 kB)
Collecting packaging (from pytest==8.3.2->-r requirements.txt (line 6))
  Using cached packaging-24.2-py3-none-any.whl (65 kB)
Collecting pluggy<2,>=1.5 (from pytest==8.3.2->-r requirements.txt (line 6))
  Using cached pluggy-1.5.0-py3-none-any.whl (20 kB)
INFO: pip is looking at multiple versions of pytest-env to determine which version is compatible with other requirements. This could take a while.
Collecting pytest-env (from -r requirements.txt (line 7))
  Using cached pytest_env-1.1.4-py3-none-any.whl (6.2 kB)
Collecting distlib<1,>=0.3.7 (from virtualenv>=20.0.8->pre-commit==2.20.0->-r requirements.txt (line 5))
  Using cached distlib-0.3.9-py2.py3-none-any.whl (468 kB)
Collecting filelock<4,>=3.12.2 (from virtualenv>=20.0.8->pre-commit==2.20.0->-r requirements.txt (line 5))
  Using cached filelock-3.16.1-py3-none-any.whl (16 kB)
Collecting platformdirs<5,>=3.9.1 (from virtualenv>=20.0.8->pre-commit==2.20.0->-r requirements.txt (line 5))
  Using cached platformdirs-4.3.6-py3-none-any.whl (18 kB)
Installing collected packages: sortedcontainers, distlib, typing_extensions, toml, pyyaml, pytest-runner, pluggy, platformdirs, packaging, numpy, nodeenv, llvmlite, iniconfig, identify, filelock, colorama, cfgv, attrs, virtualenv, pytest, numba, hypothesis, pytest-env, pre-commit
Successfully installed attrs-24.2.0 cfgv-3.4.0 colorama-0.4.3 distlib-0.3.9 filelock-3.16.1 hypothesis-6.54.0 identify-2.6.3 iniconfig-2.0.0 llvmlite-0.43.0 nodeenv-1.9.1 numba-0.60.0 numpy-2.0.0 packaging-24.2 platformdirs-4.3.6 pluggy-1.5.0 pre-commit-2.20.0 pytest-8.3.2 pytest-env-1.1.4 pytest-runner-5.2 pyyaml-6.0.2 sortedcontainers-2.4.0 toml-0.10.2 typing_extensions-4.12.2 virtualenv-20.28.0

[notice] A new release of pip is available: 23.1.2 -> 24.3.1
[notice] To update, run: pip install --upgrade pip
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pip3 install -r requirements.extra.txt
Collecting altair==4.2.2 (from -r requirements.extra.txt (line 1))
  Using cached altair-4.2.2-py3-none-any.whl (813 kB)
Collecting datasets==2.4.0 (from -r requirements.extra.txt (line 2))
  Using cached datasets-2.4.0-py3-none-any.whl (365 kB)
Collecting embeddings==0.0.8 (from -r requirements.extra.txt (line 3))
  Using cached embeddings-0.0.8-py3-none-any.whl (12 kB)
Collecting networkx==3.3 (from -r requirements.extra.txt (line 4))
  Using cached networkx-3.3-py3-none-any.whl (1.7 MB)
Collecting plotly==4.14.3 (from -r requirements.extra.txt (line 5))
  Using cached plotly-4.14.3-py2.py3-none-any.whl (13.2 MB)
Collecting pydot==1.4.1 (from -r requirements.extra.txt (line 6))
  Using cached pydot-1.4.1-py2.py3-none-any.whl (19 kB)
Requirement already satisfied: python-mnist in ./.venv/lib/python3.11/site-packages (from -r requirements.extra.txt (line 7)) (0.7)
Collecting streamlit==1.12.0 (from -r requirements.extra.txt (line 8))
  Using cached streamlit-1.12.0-py2.py3-none-any.whl (9.1 MB)
Collecting streamlit-ace (from -r requirements.extra.txt (line 9))
  Using cached streamlit_ace-0.1.1-py3-none-any.whl (3.6 MB)
Collecting torch (from -r requirements.extra.txt (line 10))
  Using cached torch-2.5.1-cp311-none-macosx_11_0_arm64.whl (63.9 MB)
Collecting watchdog==1.0.2 (from -r requirements.extra.txt (line 11))
  Using cached watchdog-1.0.2-cp311-cp311-macosx_12_0_arm64.whl
Collecting entrypoints (from altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached entrypoints-0.4-py3-none-any.whl (5.3 kB)
Collecting jinja2 (from altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached jinja2-3.1.4-py3-none-any.whl (133 kB)
Collecting jsonschema>=3.0 (from altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached jsonschema-4.23.0-py3-none-any.whl (88 kB)
Requirement already satisfied: numpy in ./.venv/lib/python3.11/site-packages (from altair==4.2.2->-r requirements.extra.txt (line 1)) (2.0.0)
Collecting pandas>=0.18 (from altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached pandas-2.2.3-cp311-cp311-macosx_11_0_arm64.whl (11.3 MB)
Collecting toolz (from altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached toolz-1.0.0-py3-none-any.whl (56 kB)
Collecting pyarrow>=6.0.0 (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached pyarrow-18.1.0-cp311-cp311-macosx_12_0_arm64.whl (29.6 MB)
Collecting dill<0.3.6 (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached dill-0.3.5.1-py2.py3-none-any.whl (95 kB)
Collecting requests>=2.19.0 (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached requests-2.32.3-py3-none-any.whl (64 kB)
Collecting tqdm>=4.62.1 (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached tqdm-4.67.1-py3-none-any.whl (78 kB)
Collecting xxhash (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached xxhash-3.5.0-cp311-cp311-macosx_11_0_arm64.whl (30 kB)
Collecting multiprocess (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached multiprocess-0.70.17-py311-none-any.whl (144 kB)
Collecting fsspec[http]>=2021.11.1 (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached fsspec-2024.10.0-py3-none-any.whl (179 kB)
Collecting aiohttp (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached aiohttp-3.11.10-cp311-cp311-macosx_11_0_arm64.whl (455 kB)
Collecting huggingface-hub<1.0.0,>=0.1.0 (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached huggingface_hub-0.26.5-py3-none-any.whl (447 kB)
Requirement already satisfied: packaging in ./.venv/lib/python3.11/site-packages (from datasets==2.4.0->-r requirements.extra.txt (line 2)) (24.2)
Collecting responses<0.19 (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached responses-0.18.0-py3-none-any.whl (38 kB)
Collecting retrying>=1.3.3 (from plotly==4.14.3->-r requirements.extra.txt (line 5))
  Using cached retrying-1.3.4-py3-none-any.whl (11 kB)
Collecting six (from plotly==4.14.3->-r requirements.extra.txt (line 5))
  Using cached six-1.17.0-py2.py3-none-any.whl (11 kB)
Collecting pyparsing>=2.1.4 (from pydot==1.4.1->-r requirements.extra.txt (line 6))
  Using cached pyparsing-3.2.0-py3-none-any.whl (106 kB)
Collecting blinker>=1.0.0 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached blinker-1.9.0-py3-none-any.whl (8.5 kB)
Collecting cachetools>=4.0 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached cachetools-5.5.0-py3-none-any.whl (9.5 kB)
Collecting click>=7.0 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached click-8.1.7-py3-none-any.whl (97 kB)
Collecting importlib-metadata>=1.4 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached importlib_metadata-8.5.0-py3-none-any.whl (26 kB)
Collecting pillow>=6.2.0 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached pillow-11.0.0-cp311-cp311-macosx_11_0_arm64.whl (3.0 MB)
Collecting protobuf<4,>=3.12 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached protobuf-3.20.3-py2.py3-none-any.whl (162 kB)
Collecting pydeck>=0.1.dev5 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached pydeck-0.9.1-py2.py3-none-any.whl (6.9 MB)
Collecting pympler>=0.9 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached Pympler-1.1-py3-none-any.whl (165 kB)
Collecting python-dateutil (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
Collecting rich>=10.11.0 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached rich-13.9.4-py3-none-any.whl (242 kB)
Collecting semver (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached semver-3.0.2-py3-none-any.whl (17 kB)
Requirement already satisfied: toml in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (0.10.2)
Collecting tornado>=5.0 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached tornado-6.4.2-cp38-abi3-macosx_10_9_universal2.whl (436 kB)
Requirement already satisfied: typing-extensions>=3.10.0.0 in ./.venv/lib/python3.11/site-packages (from streamlit==1.12.0->-r requirements.extra.txt (line 8)) (4.12.2)
Collecting tzlocal>=1.1 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached tzlocal-5.2-py3-none-any.whl (17 kB)
Collecting validators>=0.2 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached validators-0.34.0-py3-none-any.whl (43 kB)
Collecting gitpython!=3.1.19 (from streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached GitPython-3.1.43-py3-none-any.whl (207 kB)
Requirement already satisfied: filelock in ./.venv/lib/python3.11/site-packages (from torch->-r requirements.extra.txt (line 10)) (3.16.1)
Collecting sympy==1.13.1 (from torch->-r requirements.extra.txt (line 10))
  Using cached sympy-1.13.1-py3-none-any.whl (6.2 MB)
Collecting mpmath<1.4,>=1.1.0 (from sympy==1.13.1->torch->-r requirements.extra.txt (line 10))
  Using cached mpmath-1.3.0-py3-none-any.whl (536 kB)
Collecting aiohappyeyeballs>=2.3.0 (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached aiohappyeyeballs-2.4.4-py3-none-any.whl (14 kB)
Collecting aiosignal>=1.1.2 (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached aiosignal-1.3.1-py3-none-any.whl (7.6 kB)
Requirement already satisfied: attrs>=17.3.0 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2)) (24.2.0)
Collecting frozenlist>=1.1.1 (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached frozenlist-1.5.0-cp311-cp311-macosx_11_0_arm64.whl (52 kB)
Collecting multidict<7.0,>=4.5 (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached multidict-6.1.0-cp311-cp311-macosx_11_0_arm64.whl (29 kB)
Collecting propcache>=0.2.0 (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached propcache-0.2.1-cp311-cp311-macosx_11_0_arm64.whl (45 kB)
Collecting yarl<2.0,>=1.17.0 (from aiohttp->datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached yarl-1.18.3-cp311-cp311-macosx_11_0_arm64.whl (92 kB)
Collecting gitdb<5,>=4.0.1 (from gitpython!=3.1.19->streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached gitdb-4.0.11-py3-none-any.whl (62 kB)
Requirement already satisfied: pyyaml>=5.1 in ./.venv/lib/python3.11/site-packages (from huggingface-hub<1.0.0,>=0.1.0->datasets==2.4.0->-r requirements.extra.txt (line 2)) (6.0.2)
Collecting zipp>=3.20 (from importlib-metadata>=1.4->streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached zipp-3.21.0-py3-none-any.whl (9.6 kB)
Collecting jsonschema-specifications>=2023.03.6 (from jsonschema>=3.0->altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached jsonschema_specifications-2024.10.1-py3-none-any.whl (18 kB)
Collecting referencing>=0.28.4 (from jsonschema>=3.0->altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached referencing-0.35.1-py3-none-any.whl (26 kB)
Collecting rpds-py>=0.7.1 (from jsonschema>=3.0->altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached rpds_py-0.22.3-cp311-cp311-macosx_11_0_arm64.whl (349 kB)
Collecting pytz>=2020.1 (from pandas>=0.18->altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached pytz-2024.2-py2.py3-none-any.whl (508 kB)
Collecting tzdata>=2022.7 (from pandas>=0.18->altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached tzdata-2024.2-py2.py3-none-any.whl (346 kB)
Collecting MarkupSafe>=2.0 (from jinja2->altair==4.2.2->-r requirements.extra.txt (line 1))
  Using cached MarkupSafe-3.0.2-cp311-cp311-macosx_11_0_arm64.whl (12 kB)
Collecting charset-normalizer<4,>=2 (from requests>=2.19.0->datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached charset_normalizer-3.4.0-cp311-cp311-macosx_11_0_arm64.whl (118 kB)
Collecting idna<4,>=2.5 (from requests>=2.19.0->datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached idna-3.10-py3-none-any.whl (70 kB)
Collecting urllib3<3,>=1.21.1 (from requests>=2.19.0->datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached urllib3-2.2.3-py3-none-any.whl (126 kB)
Collecting certifi>=2017.4.17 (from requests>=2.19.0->datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached certifi-2024.8.30-py3-none-any.whl (167 kB)
Collecting markdown-it-py>=2.2.0 (from rich>=10.11.0->streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached markdown_it_py-3.0.0-py3-none-any.whl (87 kB)
Collecting pygments<3.0.0,>=2.13.0 (from rich>=10.11.0->streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached pygments-2.18.0-py3-none-any.whl (1.2 MB)
INFO: pip is looking at multiple versions of multiprocess to determine which version is compatible with other requirements. This could take a while.
Collecting multiprocess (from datasets==2.4.0->-r requirements.extra.txt (line 2))
  Using cached multiprocess-0.70.16-py311-none-any.whl (143 kB)
  Using cached multiprocess-0.70.15-py311-none-any.whl (135 kB)
  Using cached multiprocess-0.70.14-py310-none-any.whl (134 kB)
  Using cached multiprocess-0.70.13-py310-none-any.whl (133 kB)
Collecting smmap<6,>=3.0.1 (from gitdb<5,>=4.0.1->gitpython!=3.1.19->streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached smmap-5.0.1-py3-none-any.whl (24 kB)
Collecting mdurl~=0.1 (from markdown-it-py>=2.2.0->rich>=10.11.0->streamlit==1.12.0->-r requirements.extra.txt (line 8))
  Using cached mdurl-0.1.2-py3-none-any.whl (10.0 kB)
Installing collected packages: pytz, mpmath, zipp, xxhash, watchdog, validators, urllib3, tzlocal, tzdata, tqdm, tornado, toolz, sympy, smmap, six, semver, rpds-py, pyparsing, pympler, pygments, pyarrow, protobuf, propcache, pillow, networkx, multidict, mdurl, MarkupSafe, idna, fsspec, frozenlist, entrypoints, dill, click, charset-normalizer, certifi, cachetools, blinker, aiohappyeyeballs, yarl, retrying, requests, referencing, python-dateutil, pydot, multiprocess, markdown-it-py, jinja2, importlib-metadata, gitdb, aiosignal, torch, rich, responses, pydeck, plotly, pandas, jsonschema-specifications, huggingface-hub, gitpython, embeddings, aiohttp, jsonschema, datasets, altair, streamlit, streamlit-ace
Successfully installed MarkupSafe-3.0.2 aiohappyeyeballs-2.4.4 aiohttp-3.11.10 aiosignal-1.3.1 altair-4.2.2 blinker-1.9.0 cachetools-5.5.0 certifi-2024.8.30 charset-normalizer-3.4.0 click-8.1.7 datasets-2.4.0 dill-0.3.5.1 embeddings-0.0.8 entrypoints-0.4 frozenlist-1.5.0 fsspec-2024.10.0 gitdb-4.0.11 gitpython-3.1.43 huggingface-hub-0.26.5 idna-3.10 importlib-metadata-8.5.0 jinja2-3.1.4 jsonschema-4.23.0 jsonschema-specifications-2024.10.1 markdown-it-py-3.0.0 mdurl-0.1.2 mpmath-1.3.0 multidict-6.1.0 multiprocess-0.70.13 networkx-3.3 pandas-2.2.3 pillow-11.0.0 plotly-4.14.3 propcache-0.2.1 protobuf-3.20.3 pyarrow-18.1.0 pydeck-0.9.1 pydot-1.4.1 pygments-2.18.0 pympler-1.1 pyparsing-3.2.0 python-dateutil-2.9.0.post0 pytz-2024.2 referencing-0.35.1 requests-2.32.3 responses-0.18.0 retrying-1.3.4 rich-13.9.4 rpds-py-0.22.3 semver-3.0.2 six-1.17.0 smmap-5.0.1 streamlit-1.12.0 streamlit-ace-0.1.1 sympy-1.13.1 toolz-1.0.0 torch-2.5.1 tornado-6.4.2 tqdm-4.67.1 tzdata-2024.2 tzlocal-5.2 urllib3-2.2.3 validators-0.34.0 watchdog-1.0.2 xxhash-3.5.0 yarl-1.18.3 zipp-3.21.0

[notice] A new release of pip is available: 23.1.2 -> 24.3.1
[notice] To update, run: pip install --upgrade pip
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % python project/run_sentiment.py                             

Traceback (most recent call last):
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/project/run_sentiment.py", line 5, in <module>
    import minitorch
ModuleNotFoundError: No module named 'minitorch'
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % python project/run_sentiment.py

Traceback (most recent call last):
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/project/run_sentiment.py", line 5, in <module>
    import minitorch
ModuleNotFoundError: No module named 'minitorch'
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % tree
.
├── README.md
├── data
├── files_to_sync.txt
├── minitorch
│   ├── __init__.py
│   ├── __pycache__
│   │   ├── __init__.cpython-311.pyc
│   │   ├── autodiff.cpython-311.pyc
│   │   ├── cuda_ops.cpython-311.pyc
│   │   ├── datasets.cpython-311.pyc
│   │   ├── fast_conv.cpython-311.pyc
│   │   ├── fast_ops.cpython-311.pyc
│   │   ├── module.cpython-311.pyc
│   │   ├── nn.cpython-311.pyc
│   │   ├── operators.cpython-311.pyc
│   │   ├── optim.cpython-311.pyc
│   │   ├── scalar.cpython-311.pyc
│   │   ├── scalar_functions.cpython-311.pyc
│   │   ├── tensor.cpython-311.pyc
│   │   ├── tensor_data.cpython-311.pyc
│   │   ├── tensor_functions.cpython-311.pyc
│   │   ├── tensor_ops.cpython-311.pyc
│   │   └── testing.cpython-311.pyc
│   ├── autodiff.py
│   ├── cuda_ops.py
│   ├── datasets.py
│   ├── fast_conv.py
│   ├── fast_ops.py
│   ├── module.py
│   ├── nn.py
│   ├── operators.py
│   ├── optim.py
│   ├── scalar.py
│   ├── scalar_functions.py
│   ├── tensor.py
│   ├── tensor_data.py
│   ├── tensor_functions.py
│   ├── tensor_ops.py
│   └── testing.py
├── project
│   ├── app.py
│   ├── data
│   │   ├── t10k-images-idx3-ubyte
│   │   ├── t10k-labels-idx1-ubyte
│   │   ├── train-images-idx3-ubyte
│   │   └── train-labels-idx1-ubyte
│   ├── graph_builder.py
│   ├── interface
│   │   ├── plots.py
│   │   ├── streamlit_utils.py
│   │   └── train.py
│   ├── math_interface.py
│   ├── module_interface.py
│   ├── parallel_check.py
│   ├── run_manual.py
│   ├── run_mnist_interface.py
│   ├── run_mnist_multiclass.py
│   ├── run_scalar.py
│   ├── run_sentiment.py
│   ├── run_tensor.py
│   ├── run_torch.py
│   ├── sentiment_interface.py
│   ├── show_expression.py
│   ├── show_expression_interface.py
│   ├── show_tensor.py
│   └── tensor_interface.py
├── pyproject.toml
├── requirements.extra.txt
├── requirements.txt
├── setup.py
├── sync_previous_module.py
└── tests
    ├── __init__.py
    ├── __pycache__
    │   ├── __init__.cpython-311.pyc
    │   ├── strategies.cpython-311.pyc
    │   ├── tensor_strategies.cpython-311.pyc
    │   ├── test_conv.cpython-311-pytest-8.3.2.pyc
    │   ├── test_nn.cpython-311-pytest-8.3.2.pyc
    │   └── test_tensor_general.cpython-311-pytest-8.3.2.pyc
    ├── strategies.py
    ├── tensor_strategies.py
    ├── test_conv.py
    ├── test_nn.py
    └── test_tensor_general.py

9 directories, 77 files
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % python -m project.run_sentiment

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/project/run_sentiment.py", line 6, in <module>
    from datasets import load_dataset
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/datasets/__init__.py", line 39, in <module>
    from .builder import ArrowBasedBuilder, BeamBasedBuilder, BuilderConfig, DatasetBuilder, GeneratorBasedBuilder
  File "/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/datasets/builder.py", line 85, in <module>
    @dataclass
     ^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 1230, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 1220, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 958, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/syedakazmi-shah/.pyenv/versions/3.11.4/lib/python3.11/dataclasses.py", line 815, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'datasets.utils.version.Version'> for field version is not allowed: use default_factory
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % pip install --upgrade datasets

Requirement already satisfied: datasets in ./.venv/lib/python3.11/site-packages (2.4.0)
Collecting datasets
  Downloading datasets-3.1.0-py3-none-any.whl (480 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 480.6/480.6 kB 3.0 MB/s eta 0:00:00
Requirement already satisfied: filelock in ./.venv/lib/python3.11/site-packages (from datasets) (3.16.1)
Requirement already satisfied: numpy>=1.17 in ./.venv/lib/python3.11/site-packages (from datasets) (2.0.0)
Requirement already satisfied: pyarrow>=15.0.0 in ./.venv/lib/python3.11/site-packages (from datasets) (18.1.0)
Requirement already satisfied: dill<0.3.9,>=0.3.0 in ./.venv/lib/python3.11/site-packages (from datasets) (0.3.5.1)
Requirement already satisfied: pandas in ./.venv/lib/python3.11/site-packages (from datasets) (2.2.3)
Requirement already satisfied: requests>=2.32.2 in ./.venv/lib/python3.11/site-packages (from datasets) (2.32.3)
Requirement already satisfied: tqdm>=4.66.3 in ./.venv/lib/python3.11/site-packages (from datasets) (4.67.1)
Requirement already satisfied: xxhash in ./.venv/lib/python3.11/site-packages (from datasets) (3.5.0)
Requirement already satisfied: multiprocess<0.70.17 in ./.venv/lib/python3.11/site-packages (from datasets) (0.70.13)
Collecting fsspec[http]<=2024.9.0,>=2023.1.0 (from datasets)
  Downloading fsspec-2024.9.0-py3-none-any.whl (179 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 179.3/179.3 kB 14.9 MB/s eta 0:00:00
Requirement already satisfied: aiohttp in ./.venv/lib/python3.11/site-packages (from datasets) (3.11.10)
Requirement already satisfied: huggingface-hub>=0.23.0 in ./.venv/lib/python3.11/site-packages (from datasets) (0.26.5)
Requirement already satisfied: packaging in ./.venv/lib/python3.11/site-packages (from datasets) (24.2)
Requirement already satisfied: pyyaml>=5.1 in ./.venv/lib/python3.11/site-packages (from datasets) (6.0.2)
Requirement already satisfied: aiohappyeyeballs>=2.3.0 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets) (2.4.4)
Requirement already satisfied: aiosignal>=1.1.2 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets) (1.3.1)
Requirement already satisfied: attrs>=17.3.0 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets) (24.2.0)
Requirement already satisfied: frozenlist>=1.1.1 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets) (1.5.0)
Requirement already satisfied: multidict<7.0,>=4.5 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets) (6.1.0)
Requirement already satisfied: propcache>=0.2.0 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets) (0.2.1)
Requirement already satisfied: yarl<2.0,>=1.17.0 in ./.venv/lib/python3.11/site-packages (from aiohttp->datasets) (1.18.3)
Requirement already satisfied: typing-extensions>=3.7.4.3 in ./.venv/lib/python3.11/site-packages (from huggingface-hub>=0.23.0->datasets) (4.12.2)
Requirement already satisfied: charset-normalizer<4,>=2 in ./.venv/lib/python3.11/site-packages (from requests>=2.32.2->datasets) (3.4.0)
Requirement already satisfied: idna<4,>=2.5 in ./.venv/lib/python3.11/site-packages (from requests>=2.32.2->datasets) (3.10)
Requirement already satisfied: urllib3<3,>=1.21.1 in ./.venv/lib/python3.11/site-packages (from requests>=2.32.2->datasets) (2.2.3)
Requirement already satisfied: certifi>=2017.4.17 in ./.venv/lib/python3.11/site-packages (from requests>=2.32.2->datasets) (2024.8.30)
Requirement already satisfied: python-dateutil>=2.8.2 in ./.venv/lib/python3.11/site-packages (from pandas->datasets) (2.9.0.post0)
Requirement already satisfied: pytz>=2020.1 in ./.venv/lib/python3.11/site-packages (from pandas->datasets) (2024.2)
Requirement already satisfied: tzdata>=2022.7 in ./.venv/lib/python3.11/site-packages (from pandas->datasets) (2024.2)
Requirement already satisfied: six>=1.5 in ./.venv/lib/python3.11/site-packages (from python-dateutil>=2.8.2->pandas->datasets) (1.17.0)
Installing collected packages: fsspec, datasets
  Attempting uninstall: fsspec
    Found existing installation: fsspec 2024.10.0
    Uninstalling fsspec-2024.10.0:
      Successfully uninstalled fsspec-2024.10.0
  Attempting uninstall: datasets
    Found existing installation: datasets 2.4.0
    Uninstalling datasets-2.4.0:
      Successfully uninstalled datasets-2.4.0
Successfully installed datasets-3.1.0 fsspec-2024.9.0

[notice] A new release of pip is available: 23.1.2 -> 24.3.1
[notice] To update, run: pip install --upgrade pip
(.venv) syedakazmi-shah@Syedas-MBP mod4-manzala % python -m project.run_sentiment

README.md: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████| 35.3k/35.3k [00:00<00:00, 32.4MB/s]
train-00000-of-00001.parquet: 100%|██████████████████████████████████████████████████████████████████████████████████████████| 3.11M/3.11M [00:00<00:00, 8.51MB/s]
validation-00000-of-00001.parquet: 100%|█████████████████████████████████████████████████████████████████████████████████████| 72.8k/72.8k [00:00<00:00, 3.58MB/s]
test-00000-of-00001.parquet: 100%|█████████████████████████████████████████████████████████████████████████████████████████████| 148k/148k [00:00<00:00, 7.38MB/s]
Generating train split: 100%|███████████████████████████████████████████████████████████████████████████████████| 67349/67349 [00:00<00:00, 1186266.91 examples/s]
Generating validation split: 100%|███████████████████████████████████████████████████████████████████████████████████| 872/872 [00:00<00:00, 641654.93 examples/s]
Generating test split: 100%|███████████████████████████████████████████████████████████████████████████████████████| 1821/1821 [00:00<00:00, 971239.52 examples/s]
Downloading from http://nlp.stanford.edu/data/glove.6B.zip to /Users/syedakazmi-shah/.embeddings/glove/wikipedia_gigaword.zip
/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'nlp.stanford.edu'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
/Users/syedakazmi-shah/Documents/Cornell/MLE/mod4-manzala/.venv/lib/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'downloads.cs.stanford.edu'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
  warnings.warn(
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 400000/400000 [00:21<00:00, 18319.97it/s]
missing pre-trained embedding for 55 unknown words
Epoch 1, loss 31.20869877937119, train accuracy: 52.44%
Validation accuracy: 54.00%
Best Valid accuracy: 54.00%
Epoch 2, loss 31.02644536875389, train accuracy: 53.56%
Validation accuracy: 60.00%
Best Valid accuracy: 60.00%
Epoch 3, loss 30.839796073870005, train accuracy: 56.44%
Validation accuracy: 69.00%
Best Valid accuracy: 69.00%
Epoch 4, loss 30.549471210495607, train accuracy: 59.33%
Validation accuracy: 63.00%
Best Valid accuracy: 69.00%
Epoch 5, loss 30.33455687542565, train accuracy: 59.56%
Validation accuracy: 62.00%
Best Valid accuracy: 69.00%
Epoch 6, loss 30.107021808718812, train accuracy: 62.22%
Validation accuracy: 59.00%
Best Valid accuracy: 69.00%
Epoch 7, loss 29.80675122738262, train accuracy: 62.67%
Validation accuracy: 58.00%
Best Valid accuracy: 69.00%
Epoch 8, loss 29.361123998828777, train accuracy: 62.44%
Validation accuracy: 68.00%
Best Valid accuracy: 69.00%
Epoch 9, loss 29.36293185772052, train accuracy: 63.78%
Validation accuracy: 71.00%
Best Valid accuracy: 71.00%
Epoch 10, loss 28.76655550221041, train accuracy: 64.67%
Validation accuracy: 71.00%
Best Valid accuracy: 71.00%
Epoch 11, loss 28.520771707252887, train accuracy: 68.89%
Validation accuracy: 68.00%
Best Valid accuracy: 71.00%
Epoch 12, loss 27.748972853877042, train accuracy: 68.89%
Validation accuracy: 67.00%
Best Valid accuracy: 71.00%
Epoch 13, loss 27.265006382452242, train accuracy: 71.56%
Validation accuracy: 64.00%
Best Valid accuracy: 71.00%
Epoch 14, loss 26.64990639685923, train accuracy: 71.56%
Validation accuracy: 70.00%
Best Valid accuracy: 71.00%
Epoch 15, loss 26.241324082948143, train accuracy: 71.78%
Validation accuracy: 72.00%
Best Valid accuracy: 72.00%
Epoch 16, loss 25.7513027382742, train accuracy: 72.67%
Validation accuracy: 67.00%
Best Valid accuracy: 72.00%
Epoch 17, loss 24.8993891293429, train accuracy: 73.56%
Validation accuracy: 70.00%
Best Valid accuracy: 72.00%
Epoch 18, loss 24.34712522825554, train accuracy: 72.67%
Validation accuracy: 73.00%
Best Valid accuracy: 73.00%
Epoch 19, loss 23.653139677647715, train accuracy: 73.78%
Validation accuracy: 68.00%
Best Valid accuracy: 73.00%
Epoch 20, loss 23.352129951193753, train accuracy: 75.56%
Validation accuracy: 73.00%
Best Valid accuracy: 73.00%
Epoch 21, loss 22.197629948094576, train accuracy: 75.11%
Validation accuracy: 73.00%
Best Valid accuracy: 73.00%
Epoch 22, loss 21.997619132181757, train accuracy: 77.33%
Validation accuracy: 73.00%
Best Valid accuracy: 73.00%
Epoch 23, loss 21.433808123634027, train accuracy: 76.89%
Validation accuracy: 70.00%
Best Valid accuracy: 73.00%
Epoch 24, loss 21.003547916257084, train accuracy: 76.89%
Validation accuracy: 74.00%
Best Valid accuracy: 74.00%
Epoch 25, loss 20.184826635531188, train accuracy: 79.11%
Validation accuracy: 75.00%
Best Valid accuracy: 75.00%
Epoch 26, loss 19.907278494575788, train accuracy: 80.89%
Validation accuracy: 70.00%
Best Valid accuracy: 75.00%
Epoch 27, loss 19.44463274722161, train accuracy: 80.00%
Validation accuracy: 72.00%
Best Valid accuracy: 75.00%
Epoch 28, loss 19.418406625262143, train accuracy: 79.11%
Validation accuracy: 74.00%
Best Valid accuracy: 75.00%
Epoch 29, loss 18.078736287708757, train accuracy: 81.78%
Validation accuracy: 74.00%
Best Valid accuracy: 75.00%
Epoch 30, loss 16.951128112342136, train accuracy: 86.67%
Validation accuracy: 75.00%
Best Valid accuracy: 75.00%
Epoch 31, loss 17.44198267128977, train accuracy: 83.33%
Validation accuracy: 69.00%
Best Valid accuracy: 75.00%
Epoch 32, loss 16.725304513121444, train accuracy: 84.00%
Validation accuracy: 71.00%
Best Valid accuracy: 75.00%
Epoch 33, loss 16.746852537864704, train accuracy: 83.33%
Validation accuracy: 69.00%
Best Valid accuracy: 75.00%
Epoch 34, loss 16.415814112649134, train accuracy: 82.89%
Validation accuracy: 73.00%
Best Valid accuracy: 75.00%
Epoch 35, loss 15.318208783774367, train accuracy: 83.78%
Validation accuracy: 72.00%
Best Valid accuracy: 75.00%
Epoch 36, loss 15.20671757342295, train accuracy: 85.11%
Validation accuracy: 72.00%
Best Valid accuracy: 75.00%
Epoch 37, loss 15.739117126128024, train accuracy: 84.67%
Validation accuracy: 73.00%
Best Valid accuracy: 75.00%
Epoch 38, loss 14.799631008731772, train accuracy: 86.44%
Validation accuracy: 73.00%
Best Valid accuracy: 75.00%
Epoch 39, loss 13.792368753907125, train accuracy: 87.11%
Validation accuracy: 72.00%
Best Valid accuracy: 75.00%
Epoch 40, loss 14.068456993439469, train accuracy: 86.44%
Validation accuracy: 70.00%
Best Valid accuracy: 75.00%
Epoch 41, loss 14.084528530155648, train accuracy: 84.44%
Validation accuracy: 73.00%
Best Valid accuracy: 75.00%
Epoch 42, loss 13.219345246201017, train accuracy: 86.44%
Validation accuracy: 72.00%
Best Valid accuracy: 75.00%
Epoch 43, loss 13.93375105761123, train accuracy: 86.00%
Validation accuracy: 71.00%
Best Valid accuracy: 75.00%
Epoch 44, loss 13.401574949127879, train accuracy: 87.33%
Validation accuracy: 69.00%
Best Valid accuracy: 75.00%
Epoch 45, loss 12.860127706417192, train accuracy: 86.89%
Validation accuracy: 69.00%
Best Valid accuracy: 75.00%
Epoch 46, loss 12.794752081144765, train accuracy: 85.56%
Validation accuracy: 69.00%
Best Valid accuracy: 75.00%
Epoch 47, loss 13.287227342027636, train accuracy: 85.78%
Validation accuracy: 70.00%
Best Valid accuracy: 75.00%
Epoch 48, loss 12.451647820151148, train accuracy: 85.56%
Validation accuracy: 69.00%
Best Valid accuracy: 75.00%
Epoch 49, loss 12.803619060430927, train accuracy: 84.89%
Validation accuracy: 70.00%
Best Valid accuracy: 75.00%
